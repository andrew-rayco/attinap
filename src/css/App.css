body {
    display: flex;
    justify-content: center;
    padding: 1em;
    background: hotpink;
}

.main {
    display: flex;
    flex-direction: column;
    align-items: center;
}

button {
    padding: 1em;
    border: 4px solid green;
    background: lightgreen;
    color: #666;
}

button:hover {
    border: 4px solid lightgreen;
    background: green;
    color: white;
    transition: all 0.5s;
    cursor: pointer;
}
