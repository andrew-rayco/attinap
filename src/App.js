import React, { Component } from 'react'
import moment from 'moment'

import Button from './components/Button'

import './css/App.css'

class App extends Component {
    state = {
        sleepTime: '',
        awakeTime: ''
    }

    updateTime(time) {
        const now = new Date()

        this.setState({
            [time]: moment(now)
        })
    }

    formatTime(time) {
        return time
            ? moment(time).format('dddd, MMMM Do YYYY, h:mm:ss a')
            : 'No time yet'
    }

    formatAgo(time) {
        return time ? moment(time).fromNow() : ''
    }

    render() {
        const { sleepTime, awakeTime } = this.state

        return (
            <div className="main">
                <Button
                    time={this.formatTime(sleepTime)}
                    updateTime={() => this.updateTime('sleepTime')}
                    title={'Just went to sleep'}
                    ago={this.formatAgo(sleepTime)}
                    className={'sleep-time'}
                />
                <Button
                    time={this.formatTime(awakeTime)}
                    updateTime={() => this.updateTime('awakeTime')}
                    title={'Just woke up'}
                    ago={this.formatAgo(awakeTime)}
                    className={'awake-time'}
                />
            </div>
        )
    }
}

export default App
