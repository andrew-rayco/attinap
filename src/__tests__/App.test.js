import React from 'react'
import { render } from '@testing-library/react'
import App from '../App'

test('App snapshot', () => {
    const { container } = render(<App />)

    expect(container.firstChild).toMatchSnapshot()
})

test('Renders both Button components', () => {
    const { container, getByText } = render(<App />)

    const sleepButtonNode = getByText('Just went to sleep')
    const awakeButtonNode = getByText('Just woke up')

    expect(sleepButtonNode).toBeTruthy()
    expect(awakeButtonNode).toBeTruthy()
})
