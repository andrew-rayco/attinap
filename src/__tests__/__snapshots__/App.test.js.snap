// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<div
  class="main"
>
  <h1>
    AttiNap
  </h1>
  <p>
    Currently:
  </p>
  <h2
    data-testid="wake-status"
  >
    He's awake
  </h2>
  <div
    class="btn sleep-time"
  >
    <button>
      Just went to sleep
    </button>
    <p
      class="light"
      data-testid="time"
    >
      No time yet today
    </p>
    <p>
      <span>
         
      </span>
    </p>
    <div
      class="manage-links"
    >
      <a
        href="#"
      >
        Undo last
      </a>
      |
      <a
        href="#"
      >
        Add time
      </a>
    </div>
  </div>
  <div
    class="btn awake-time"
  >
    <button>
      Just woke up
    </button>
    <p
      class="light"
      data-testid="time"
    >
      No time yet today
    </p>
    <p>
      <span>
         
      </span>
    </p>
    <div
      class="manage-links"
    >
      <a
        href="#"
      >
        Undo last
      </a>
      |
      <a
        href="#"
      >
        Add time
      </a>
    </div>
  </div>
</div>
`;
